src/test/scala/com/fulcrumgenomics/fastq/DemuxFastqsTest.scala

```diff
@@ -516,70 +516,111 @@ class DemuxFastqsTest extends UnitSpec with OptionValues with ErrorLogLevel {
+        fastqs2.foreach { fastq => fastq.readNumber.isEmpty shouldBe true }
+        // Read names should match
+        fastqs1.map(_.header) should contain theSameElementsInOrderAs fastqs2.map(_.header)
+      }
 
-    it should s"${if (illuminaStandards) "not " else ""}append /1 and /2 to read pairs when outputting FASTQs" in {
-      val output     = outputDir()
-      val metrics    = makeTempFile("metrics", ".txt")
-      val structures = Seq(ReadStructure("17B100T"), ReadStructure("117T"))
-      new DemuxFastqs(inputs=Seq(illuminaReadNamesFastqPath, illuminaReadNamesFastqPath), output=output, metadata=sampleSheetPath,
-        readStructures=structures, metrics=Some(metrics), maxMismatches=2, minMismatchDelta=3,
-        outputType=Some(OutputType.Fastq), illuminaStandards=illuminaStandards).execute()
-
-      val sampleInfos = toSampleInfos(structures)
-
-      // Check the BAMs
-      sampleInfos.foreach { sampleInfo =>
-        val sample     = sampleInfo.sample
-        val prefix     = toSampleOutputPrefix(sample, isUnmatched=sampleInfo.isUnmatched, illuminaStandards=illuminaStandards, output, UnmatchedSampleId)
-        val extensions = FastqRecordWriter.extensions(pairedEnd=true, illuminaStandards=illuminaStandards)
-
-        def toHeader(rec: FastqRecord): String = {
-          if (illuminaStandards) rec.header.endsWith(" ReadNum:FilterFlag:0:SampleNumber") shouldBe true
-          else if (rec.readNumber.value == 1) rec.header.endsWith("/1") shouldBe true
-          else  rec.header.endsWith("/2") shouldBe true
-          rec.header.split(" ").head.replaceAll(".*:", "").replaceAll("/[12]$", "")
+      // Check the sample barcode is in the comment
+      if (fastqStandards.includeSampleBarcodes) {
+        fastqs1.map(ReadInfo(_).sampleInfo) should contain theSameElementsInOrderAs barcodes
+      }
+      else {
+        fastqs1.foreach { fastq =>
+          fastq.comment.value.endsWith("SampleNumber") shouldBe true
         }
+      }
 
-        val headersR1 = {
-          val fastq = PathUtil.pathTo(s"${prefix}${extensions.head}")
-          FastqSource(fastq).toSeq.map(toHeader).toList
-        }
-        val headersR2 = {
-          val fastq = PathUtil.pathTo(s"${prefix}${extensions.last}")
-          FastqSource(fastq).toSeq.map(toHeader).toList
-        }
+      // Check the assignment, which we encoded as the last field in the read name
+      val observedAssignments = fastqs1.map { fastq =>
+        fastq.name.replaceAll(".*:", "").replaceAll("/[12]$", "")
+      }
+      if (fastqStandards.includeSampleBarcodes) {
+        observedAssignments should contain theSameElementsInOrderAs assignments
+      }
+      else {
+        observedAssignments should contain theSameElementsInOrderAs barcodes
+      }
+   }
+  }
 
-        headersR1 should contain theSameElementsInOrderAs headersR2
+  it should "demultiplex with --fastqs-include-read-numbers=false --fastqs-includ-sample-barcodes=false" in {
```

Comment:
Fix with search/replace here and below.
```suggestion
  it should "demultiplex with --fastqs-include-read-numbers=false --fastqs-include-sample-barcodes=false" in {
```